presto-hive/src/test/java/com/facebook/presto/hive/TestDynamicBucketPruning.java

```diff
@@ -0,0 +1,141 @@
+import com.facebook.presto.common.type.StandardTypes;
+import com.facebook.presto.hive.metastore.Column;
+import com.facebook.presto.hive.metastore.Storage;
+import com.facebook.presto.hive.metastore.StorageFormat;
+import com.facebook.presto.spi.ConnectorId;
+import com.facebook.presto.spi.ConnectorPageSource;
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.spi.SplitContext;
+import com.facebook.presto.spi.TableHandle;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import static com.facebook.presto.common.type.IntegerType.INTEGER;
+import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.expressions.LogicalRowExpressions.TRUE_CONSTANT;
+import static com.facebook.presto.hive.BucketFunctionType.HIVE_COMPATIBLE;
+import static com.facebook.presto.hive.CacheQuotaRequirement.NO_CACHE_REQUIREMENT;
+import static com.facebook.presto.hive.HiveColumnHandle.ColumnType.REGULAR;
+import static com.facebook.presto.hive.HiveQueryRunner.HIVE_CATALOG;
+import static com.facebook.presto.hive.HiveTestUtils.ROW_EXPRESSION_SERVICE;
+import static com.facebook.presto.hive.HiveTestUtils.TYPE_MANAGER;
+import static com.facebook.presto.hive.HiveTestUtils.createTestHdfsEnvironment;
+import static com.facebook.presto.hive.HiveTestUtils.getDefaultHiveBatchPageSourceFactories;
+import static com.facebook.presto.hive.HiveTestUtils.getDefaultHiveRecordCursorProvider;
+import static com.facebook.presto.hive.HiveTestUtils.getDefaultHiveSelectivePageSourceFactories;
+import static com.facebook.presto.hive.HiveType.HIVE_INT;
+import static com.facebook.presto.hive.TestHivePageSink.getColumnHandles;
+import static com.facebook.presto.hive.TestHivePageSink.getSession;
+import static com.facebook.presto.spi.schedule.NodeSelectionStrategy.NO_PREFERENCE;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static org.testng.Assert.assertEquals;
+
+public class TestDynamicBucketPruning
+{
+    private static final String SCHEMA_NAME = "test";
+    private static final String TABLE_NAME = "test";
+    private static final Column BUCKET_COLUMN = new Column("l_orderkey", HIVE_INT, Optional.empty());
+
+    @Test
+    public void testDynamicBucketPruning()
+            throws Exception
+    {
+        HiveClientConfig config = new HiveClientConfig();
+        MetastoreClientConfig metastoreClientConfig = new MetastoreClientConfig();
+        HiveTransactionHandle transaction = new HiveTransactionHandle();
+        File outputFile = new File("");
+        ConnectorPageSource pageSource = createPageSource(transaction, config, new SplitContext(false, getTupleDomain()), metastoreClientConfig, outputFile);
+        assertEquals(pageSource.getClass(), HiveEmptySplitPageSource.class);
```

Comment:
Add one more assertion where `getTupleDomain()` is the one to scan. This is to cover both HiveEmptySplitPageSource and normal PageSource paths